app/controllers/ops_controller/diagnostics.rb

```diff
@@ -736,14 +736,25 @@ def diagnostics_set_form_vars
     end
   end
 
-  def database_details
+  def row_data(label, value)
+    {:cells => {:label => label, :value => value}}
+  end
+
+  def database_tab_summary
     @database_details = ActiveRecord::Base.configurations[Rails.env]
     @database_display_name =
       if @database_details["host"].in?([nil, "", "localhost", "127.0.0.1"])
         _("Internal Database")
       else
         _("External Database")
       end
+    @data = {:title => _('Basic Information')}
+    @data[:rows] = [
+      row_data(_('Name'), @database_display_name),
+      row_data(_('Hostname'), @database_details["host"]),
+      row_data(_('Database name'), @database_details["database"]),
+      row_data(_('Username'), @database_details["username"]),
+    ]
```

Comment:
similar suggestion to move them to ops_helper.rb. 
we can also reduce the usage of instance variable usage in partial files